packages/runtime/container-runtime/src/summary/summarizer.ts

```diff
@@ -46,6 +54,154 @@ import { SummarizeResultBuilder } from "./summaryGenerator.js";
 
 const summarizingError = "summarizingError";
 
+/**
+ * @legacy
+ * @alpha
+ */
+export interface ISummaryBaseConfiguration {
```

Comment:
I would suggest moving the interfaces, consts and other definitions to a `summaryDefinitions.ts` file. It doesn't really belong in `summarizer.ts` which only contians a piece of the summary related logic and doesn't even interact with container runtime that uses most of these. Actually, I think `summarizerTypes.ts` may be a better place since it already contains other summary related definitions.

Similarly, the free helper functions can move to a `summaryHelpers.ts` file.

Reply:
Changes applied, let me know if you think any other moves are worth making.